internal/internal_task_handlers.go

```diff
@@ -976,7 +976,10 @@ func (w *workflowExecutionContextImpl) retryLocalActivity(lar *localActivityResu
 			}
 
 			lar.task.attempt++
-			w.laTunnel.sendTask(lar.task)
+
+			if !w.laTunnel.sendTask(lar.task) {
```

Comment:
Consider changing this to positive assertion:
```
if w.laTunnel.sendTask(..) {
    lar.task.attempt++
}
```

Reply:
I am not sure about this. If the processing logic uses the attempt field, then changing its value after the task is sent will lead to a race condition I think.